### PR TITLE
adding examples for app auth

### PR DESCRIPTION
**Pull request by patalwell**
@gmlewis opened a seperate PR from master, have a headache from trying to revert fetches I had pulled. I'd have to cherry pick commits and would rather start fresh.

To recap, this is a simple change to README.md and an example for GitHub app authentication flow so folks can understand how to leverage installations and access tokens.

### CODE REVIEW

File: example/newfilewithappauth/main.go
Comment (gmlewis) [inline example/newfilewithappauth/main.go:8]: For lines 1-8, please add a single space after the `//`.

File: example/newfilewithappauth/main.go
Comment (gmlewis) [inline example/newfilewithappauth/main.go:33]: All these failures should use `log.Fatalf` instead of `fmt.Printf`.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @patalwell !
Just a few minor tweaks, please.

**Comment by patalwell** [comment]
@gmlewis good to go!

**Comment by gmlewis** [comment]
I'm not seeing any changes since my last review... did you push them?

Also, did you run `go test ./...` and `go mod tidy` ?
It looks like your new dependency was not added to `go.mod`... so these commands should help:

* `go get github.com/bradleyfalzon/ghinstallation`
* `go test ./...`
* `go mod tidy`

**Comment by patalwell** [comment]
@gmlewis oops, wrong branch. Pushed to the right one this time. Made the logging changes.

I didn't want to run mod tidy since we ran into issues last time with the go.mod file and go.sum files, BUT ran the commands at your discretion :)

Ran a test as well, I originally never ran one since I didn't have a test file. Also ran a linter! there was an error with one of the imports, so ran `golangci-lint run --fix `

File: go.mod
Comment (gmlewis) [inline go.mod:15]: Oh!  I think I understand now... this is being pulled in because of the new dependency of `github.com/bradleyfalzon/ghinstallation` on line 4.

Hmmm... I'm not sure we actually want to do this in that case.  Let me think about it.

Comment (gmlewis) [reply go.mod:15]: Instead of adding this example to this repo, what would you think of adding it to the `github.com/bradleyfalzon/ghinstallation` repo?

I'm wondering if that would make more sense than adding it here, since this example is very specific to that package.

Comment (patalwell) [reply go.mod:15]: Ah the woes of modularity! Wondering if we can update the version `github.com/bradleyfalzon/ghinstallation` uses to V41 https://github.com/bradleyfalzon/ghinstallation/blob/master/go.mod#L1. Right now it looks like they're using version 40. Would it be ok that we upgraded to version 40?

Comment (patalwell) [reply go.mod]: Looks like it resolved v39 instead of 40, but at least thats better than v29.

**Review by gmlewis — APPROVED** [review]
Thank you, @patalwell !
LGTM.

Awaiting second LGTM from any other contributor to this repo before merging.

**Comment by patalwell** [comment]
> Thank you, @patalwell ! LGTM.
> 
> Awaiting second LGTM from any other contributor to this repo before merging.

@wesleimp mind taking a look at these changes?

**Comment by patalwell** [comment]
Happy New Year! @gmlewis 

Hey so dont want to lose any momentum here, anyway we can ping another commiter for review?

**Comment by gmlewis** [comment]
Happy New Year to you too, @patalwell !

Maybe @Parker77 has time to review.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
Thank you, @Parker77 !
Merging.

**Pull request merged**
